cmd/virt-chroot/tap-device-maker.go

```diff
@@ -65,11 +75,21 @@ func NewCreateTapCommand() *cobra.Command {
 				return fmt.Errorf("could not parse tap device group: %v", err)
 			}
 
-			if err := createTapDevice(tapName, uint(uid), uint(gid), int(queueNumber), int(mtu)); err != nil {
-				return fmt.Errorf("failed to create tap device named %s. Reason: %v", tapName, err)
-			}
-
-			return nil
+			return createTapDevice(tapName, uint(uid), uint(gid), int(queueNumber), int(mtu))
 		},
 	}
 }
+
+func retry(retryAttempts uint, f func() error) (uint, error) {
+	var errorsString []string
+	for attemptID := uint(0); attemptID < retryAttempts; attemptID++ {
+		if err := f(); err != nil {
+			errorsString = append(errorsString, fmt.Sprintf("[%d]: %v", attemptID, err))
+			time.Sleep(time.Second)
+		} else {
+			return attemptID, nil
+		}
+	}
+
+	return retryAttempts, fmt.Errorf(strings.Join(errorsString, "\n"))
```

Comment:
For grepping logs is better to use different separator like "|" it will be like  `failed attemp 1: failed foo: failed bar | failed attempt 2: failed foo: failed bar`

Reply:
The intention was to print multiple errors, so the newline will be able to show them all if grepped. But in reality, I think the wrapper caller will be grepped, not this low level one.